### PR TITLE
Support only OTP 21 (drop support for OTP 20)

### PR DESCRIPTION
**Pull request by lucafavatellawork**
Depends on #28 
Depends on https://github.com/opensparkl/sse_cli/pull/40

Notes:
- Keep usage of `error_logger` module, rather than moving to `logger` module. Consideration of such effort is delayed to when any logging infrastructure change needed - if anytime.
    From https://erlang.org/doc/man/error_logger.html
    > In Erlang/OTP 21.0, a new API for logging was added. The old error_logger module can still be used by legacy code, but log events are redirected to the new Logger API. New code should use the Logger API directly. 

TODO:
- [x] Prepare CI of sse_cli repo to be able to support drop of OTP 20.
  - Move CI of sse_cli to Docker (from native).
  - See https://github.com/opensparkl/sse_cli/pull/40
- [x] Review usage of `erlang:get_stacktrace()` in `svc_expr_protocol:new_request_fun`. There are no evident exceptions in the process, and the tests (eunit, ct, and sse_cli) do not exercise such code path.
  - Remove - even if no test.
- [x] Review usage of `erlang:get_stacktrace()` in `svc_rest_service:handle_cast`. There are no evident exceptions in the process, and the tests (eunit, ct, and sse_cli) do not exercise such code path.
  - Remove - even if no test.
- [x] Review usage of `erlang:get_stacktrace()` in `prog_api:transaction`. There are no explicit exceptions in the code though there is a preceding call to `mnesia:transaction`.
  - Keep.

### CODE REVIEW

**Comment by lucafavatellawork** [comment]
Rebased on master following merge of #28.

File: apps/sse_listen/src/sse_listen_handler.erl [hunk 1]
```diff
@@ -311,11 +311,11 @@ notify_handler(Ref, EvmEvent) ->
             ok
         end
       catch
-        C:E ->
+        C:E:S ->
```
Comment (jacobythwaites) [inline apps/sse_listen/src/sse_listen_handler.erl]: New format, noted.

File: apps/sse_svc/src/sse_svc_dispatcher.erl [hunk 2]
```diff
@@ -81,26 +81,26 @@ dispatch_notify(
       sync_notify(
         DataEvent)
   catch
-    ?BADMATCH({error, Reason}) ->
+    ?BADMATCH({error, Reason}):StackTrace ->
       ErrorEvent =
         sse_svc_protocol_event:new_error(
-          DataEvent, Reason),
+          DataEvent, Reason, StackTrace),
```
Comment (jacobythwaites) [inline apps/sse_svc/src/sse_svc_dispatcher.erl]: Nice

File: make_app.mk [hunk 3]
```diff
@@ -350,7 +350,7 @@ lint: EXTRA_ARGS := --no_check_plt --quiet
 lint: check_otp_release $(DIALYZER_PLT)
 	@dialyzer $(PLT_ARG) $(INC_ARGS) $(EXTRA_ARGS) $(SRC_ARGS)
 
-$(DIALYZER_PLT): PLT_APPS := erts kernel compiler debugger stdlib public_key mnesia crypto inets ssl asn1 wx runtime_tools hipe syntax_tools
+$(DIALYZER_PLT): PLT_APPS := erts kernel compiler debugger stdlib public_key mnesia crypto inets ftp tftp ssl asn1 wx runtime_tools hipe syntax_tools
```
Comment (jacobythwaites) [inline make_app.mk]: Remind me why (I remember you mentioned ftp & tftp apps)

Comment (lucafavatellawork) [reply make_app.mk]: The generation of the PLT warns (harmless) about unknown functions in these apps. Hence I silenced those. I dislike including unneeded apps, but this seems the current guideline (see inclusion of wx...).

Details are in message of commit https://github.com/opensparkl/sse_core/pull/72/commits/47e42dd908be05c6e7a803cf9f4c0bfa587a57f9

**Pull request merged**